src/yoroi-wallets/cardano/utxoManager.ts

```diff
@@ -75,3 +77,61 @@ export const makeUtxoStorage = (storage: typeof storageLegacy, storagePath: stri
 
   return utxoStorage
 }
+
+const makeUtxoManagerStorage = (storage: typeof storageLegacy, storagePath: string) => {
+  const addrCounterPath = `${storagePath}/addrCounter`
+  return {
+    addrCounter: {
+      save: (addrCounter: number) => storage.write(addrCounterPath, addrCounter),
+      clear: async () => {
+        await storage.remove(addrCounterPath)
+      },
+      read: async () => (await storage.read<number>(addrCounterPath)) ?? 0,
+    },
+  } as const
+}
+
+export const makeUtxoManager = async (id: string, apiUrl: string, storage: typeof storageLegacy) => {
```

Comment:
wrapped the UtxoService + UtxoStorage within the UtxoManager, exposing only what is used by wallet